device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -349,33 +332,43 @@ public void sendReportedProperties(Set<Property> reportedProperties) throws IOEx
      * @param version the Reported property version. Cannot be negative.
      *
      * @throws IOException if called when client is not opened or called before starting twin.
-     * @throws IllegalArgumentException if reportedProperties is null or empty.
+     * @throws IllegalArgumentException if reportedProperties is null or empty or if version is negative
      */
     public void sendReportedProperties(Set<Property> reportedProperties, int version) throws IOException, IllegalArgumentException
     {
-        verifyRegisteredIfMultiplexing();
-
-        if (this.twin == null)
-        {
-            throw new IOException("Start twin before using it");
-        }
+        this.sendReportedProperties(reportedProperties, version, null, null, null, null);
+    }
 
-        if (!this.deviceIO.isOpen())
-        {
-            throw new IOException("Open the client connection before using it.");
-        }
+    /**
+     * Sends reported properties
+     * @param reportedPropertiesParameters Container for the reported properties parameters
+     * @throws IOException if called when client is not opened or called before starting twin.
+     * @throws IllegalArgumentException if reportedProperties is null or empty or if version specified in {#reportedPropertiesParameters} is negative
+     */
+    public void sendReportedProperties(ReportedPropertiesParameters reportedPropertiesParameters) throws IOException, IllegalArgumentException
```

Comment:
In order to support varying permutations of this method I opted to create a class that contains all of the parameters required.

This is just how Java does things.